data-model.d.ts

```diff
@@ -287,4 +287,24 @@ export interface DataFactory<OutQuad extends BaseQuad = Quad, InQuad extends Bas
      * @see Quad
      */
     quad(subject: InQuad['subject'], predicate: InQuad['predicate'], object: InQuad['object'], graph?: InQuad['graph']): OutQuad;
+
+    /**
+     * @param original The original term.
+     * @return A new instance of the term such that newTermInstance.equals(original) returns true.
+     * @see Term
+     */
+    fromTerm<T extends NamedNode>(original: T): NamedNode;
+    fromTerm<T extends BlankNode>(original: T): BlankNode;
+    fromTerm<T extends Literal>(original: T): Literal;
+    fromTerm<T extends Variable>(original: T): Variable;
+    fromTerm<T extends DefaultGraph>(original: T): DefaultGraph;
+    fromTerm<T extends Quad>(original: T): OutQuad;
```

Comment:
`BaseQuad` input should also presumably be supported since `BaseQuad`s are valid `Term`s